plugins/woocommerce/tests/api-core-tests/global-setup.js

```diff
@@ -1,8 +1,200 @@
+				if ( error.response ) {
+					console.log( error.response.data );
+				}
+				throw new Error( error.message );
+			} );
+			const releaseWithTagName = response.data.find(
+				( { tag_name } ) => tag_name === UPDATE_WC
+			);
+			if ( ! releaseWithTagName ) {
+				throw new Error(
+					`No release with tag_name="${ UPDATE_WC }" found. If "${ UPDATE_WC }" is a draft release, make sure to specify a GITHUB_TOKEN environment variable.`
+				);
+			}
+			const wcZipAsset = releaseWithTagName.assets.find( ( { name } ) =>
+				name.match( /^woocommerce(-trunk-nightly)?\.zip$/ )
+			);
+			if ( wcZipAsset ) {
+				return GITHUB_TOKEN
+					? wcZipAsset.url
+					: wcZipAsset.browser_download_url;
+			}
+			throw new Error(
+				`WooCommerce release with tag "${ UPDATE_WC }" found, but does not have a WooCommerce ZIP asset.`
+			);
+		};
+
+		const url = await getWCDownloadURL();
+		const params = { url };
+
+		if ( GITHUB_TOKEN ) {
+			params.authorizationToken = GITHUB_TOKEN;
+		}
+
+		const woocommerceZipPath = await downloadZip( params );
+
+		let adminLoggedIn = false;
+		let pluginActive = false;
+
+		console.log( '--------------------------------------' );
+		console.log( 'Running daily tests, resetting site...' );
+		console.log( '--------------------------------------' );
+
+		const adminRetries = 5;
+		for ( let i = 0; i < adminRetries; i++ ) {
+			try {
+				console.log( 'Trying to log-in as admin...' );
+				await setupPage.goto( '/wp-admin' );
+				await setupPage
+					.locator( 'input[name="log"]' )
+					.fill( process.env.USER_KEY );
+				await setupPage
+					.locator( 'input[name="pwd"]' )
+					.fill( process.env.USER_SECRET );
+				await setupPage.locator( 'text=Log In' ).click();
+
+				await expect( setupPage.locator( 'div.wrap > h1' ) ).toHaveText(
+					'Dashboard'
+				);
+				console.log( 'Logged-in as admin successfully.' );
+				adminLoggedIn = true;
+				break;
+			} catch ( e ) {
+				console.log(
+					`Admin log-in failed, Retrying... ${ i }/${ adminRetries }`
+				);
+				console.log( e );
+			}
+		}
+
+		if ( ! adminLoggedIn ) {
+			console.error(
+				'Cannot proceed api test, as admin login failed. Please check if the test site has been setup correctly.'
+			);
+			process.exit( 1 );
+		}
+
+		await setupPage.goto( 'wp-admin/plugins.php' );
+		await expect( setupPage.locator( 'div.wrap > h1' ) ).toHaveText(
+			'Plugins'
+		);
+
+		console.log( 'Deactivating WooCommerce Plugin...' );
+		await setupPage.locator( '#deactivate-woocommerce' ).click();
+		await expect( setupPage.locator( 'div#message' ) ).toHaveText(
+			'Plugin deactivated.Dismiss this notice.'
+		);
+
+		console.log( 'Deleting WooCommerce Plugin...' );
+		setupPage.on( 'dialog', ( dialog ) => dialog.accept() );
+		await setupPage.locator( '#delete-woocommerce' ).click();
+		await expect( setupPage.locator( '#woocommerce-deleted' ) ).toHaveText(
+			'WooCommerce was successfully deleted.'
+		);
+
```

Comment:
As mentioned in the PR description, deactivating and deleting WC will also remove all of the WC-related data/settings.  So we're doing that here.